### DISCUSSION TITLE
Remotes url changing by enviroment

### DISCUSSION THREAD

**Discussion by kod-man** [originjs/vite-plugin-federation#592] [General]
I tried to make some investigation to change 'http://localhost:5173/assets/remoteEntry.js', by env.

QA env: the link should be https// qa.sth.example.com
Test env: the link should be https// test.sth.example.com

How can I manage it? 

Ps: we use Octopus for deploying the app

```

import federation from '@originjs/vite-plugin-federation';
import react from '@vitejs/plugin-react';
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [
    react(),
    federation({
      name: 'adminPortal',
      remotes: {
        mainApp:
          '{linkToBeChanged}/assets/remoteEntry.js',
      },
      shared: ['react', 'react-dom'],
    }),
  ],
  build: {
    modulePreload: false,
    target: 'esnext',
    minify: false,
    cssCodeSplit: false,
    outDir: 'build/children1',
  },
  test: {
    environment: 'jsdom',
    globals: true,
    css: true,
    setupFiles: './src/tests/setup.js',
  },
});
```

**Comment by shivangi-sinha-maersk** [comment]
You can try this approach:

1. update scripts in package.json to pass env as mode during build
`"build:<env>": "vite build --mode <env>"` // env=test
2. In your vite config, read the mode
`const env = loadEnv(mode, process.cwd(), "");`
3. Maintain a map of all urls agianst the env keys
```
{
   <env>: {
        spot: {
             remoteApp: "app1",
             url: "<app1_Url>/assets/remoteEntry.js"
       }
   }
}
```
4. pass the remote urls using a utility function, that returns the url map based on env:
```
export default defineConfig(({ mode }) => {
  const env = loadEnv(mode, process.cwd(), "");
  return {
    base: "/",
    server: {
      port: 8080
    },
    plugins: [
      federation({
        name: "adminPortal",
        remotes: getRemotes(env),
        shared: ['react', 'react-dom'],
      }),
      
// other configs
```

**Reply by FaizanMuhammadSE** [reply]
@shivangi-sinha-maersk, your proposed solution will work but eventually create **different** builds for each environment...

I want to **keep** single build and switch **remote-urls** on runtime depending on environment...

Do you have any solution in your mind for that ?

**Comment by FaizanMuhammadSE** [comment]
@kod-man

I want to keep single build and switch remote-urls on runtime depending on environment...

Do you have any solution in your mind for that ?

**Reply by nivoset** [reply]
id preferably be able to:
```
const remote = () => moduleFederation("http://url...")
```

Then we just load it based on the string passed in

currently ive pondered just defining them all in our app and replacing the remote function based on the environment

or making it somehow load in build from the application.yaml (weee spring boot)
